### PR TITLE
[SMALLFIX] Add equals, hashcode, and toString methods

### PR DESCRIPTION
**Pull request by aaudiber**
The hashcode methods for classes with no fields return 0. We could alternately return Objects.hash(this.getClass()), but this is more complicated and less performant, and it doesn't make sense to place these classes in large hash tables.

### CODE REVIEW

File: core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java [hunk 1]
```diff
@@ -53,12 +53,28 @@ public CompleteFileOptions setUfsLength(long ufsLength) {
...
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(mUfsLength);
+  }
+
   @Override
   public String toString() {
-    return Objects.toStringHelper(this).add("ufsLength", mUfsLength).toString();
+    return Objects.toStringHelper(this)
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java]: is this how your IDE auto-formats the code?

**Comment by jsimsa** [comment]
@aaudiber thank you very much for adding these! please fix compilation issue, otherwise LGTM

File: core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java (hunk 1 above)
Comment (aaudiber) [inline core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java]: I'm doing it this way because it makes it easier to scan visually when more fields are added. The IDE formatting always scrunches up code like this. If you feel strongly I'd be fine with moving the cutoff for abandoning IDE-style to 2+ fields.

Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java]: sgtm

File: core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java [hunk 2]
```diff
@@ -53,12 +53,28 @@ public CompleteFileOptions setUfsLength(long ufsLength) {
...
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof CompleteFileOptions)) {
+      return false;
+    }
+    CompleteFileOptions that = (CompleteFileOptions) o;
+    return Objects.equal(mUfsLength, that.mUfsLength);
```
Comment (ifcharming) [inline core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java]: Can you please add some simple unit tests? such as a corner case: (o == null) && (this != null)

**Comment by ifcharming** [comment]
@aaudiber  thank you for the PR! I left a minor comment.

**Comment by jsimsa** [comment]
LGTM

**Comment by gpang** [comment]
Could you resolve the conflicts. Thanks!

File: core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java (hunk 2 above)
Comment (aaudiber) [inline core/client/src/main/java/alluxio/client/file/options/CompleteFileOptions.java]: This case works because (null instanceof X) is always false. It will take some time to cover all of these methods, do you mind if I do so in a followup PR?
